### PR TITLE
Lock down multicodec table.csv until there is governance for it

### PR DESCRIPTION
**Pull request by darobin**
### Summary

At the 2022-11-10 [IPFS Implementers Sync](https://www.notion.so/pl-strflt/IPFS-Implementers-Sync-2022-11-10-fefb941ec00b4211ba71ca571a0ec883), the decision was made to lock the multicodec `table.csv` until we design some governance for it.

This change removes push access from the Go and JS Teams.

### Why do you need this?

We are working towards standardising multi* and that includes standardising a registry from that table. We need to avoid the proliferation of registrations, and ideally carry out some amount of cleanup, and build up governance for it. The first step is to lock it down. Admins can still make changes if absolutely necessary.

### What else do we need to know?

This is my first change here, so it may be wrong, and I am new to the community which means I may be committing a woefully inappropriate faux-pas. #YOLO

**DRI:** myself

### Reviewer's Checklist

- [x] It is clear where the request is coming from (if unsure, ask)
- [x] All the automated checks passed
- [x] The YAML changes reflect the summary of the request
- [x] The Terraform plan posted as a comment reflects the summary of the request

### CODE REVIEW

**Review by rvagg — APPROVED** [review]
yeah, this is pretty minimal, admin+stewards is already an overly-broad group so this isn't even locking down much, and tbh it's not really going to change how the repo has been managed for the last couple of years.

**Review by vmx — APPROVED** [review]
It's good to see that the discussion about governance is starting, it's something that I wanted to see for years.

**Comment by galargh** [comment]
FYI, only members of [github-mgmt stewards](https://github.com/multiformats/github-mgmt/blob/01e502edd00611e683016f479ead48e3537d1a58/github/multiformats.yml#L1536) team can merge the PRs in this repo. Since there are 2 approves here already, I'm going to do it.

**Comment by darobin** [comment]
On 22/11/2022 00:05, Rod Vagg wrote:
> yeah, this is pretty minimal, admin+stewards is already an overly-broad 
> group so this isn't even locking down much, and tbh it's not really 
> going to change how the repo has been managed for the last couple of years.
Yeah — I tried to think about better options. I don't know the space 
well enough to trim down admin+stewards myself just yet. One alternative 
is to lock this down to just one person or to a small list, and possibly 
also add a mention in the readme that the table is frozen for now. The 
repo doesn't get updated that frequently, so it's manageable.

-- 
Robin Berjon
Governance & Standards at Protocol Labs
https://berjon.com/

**Comment by willscott** [comment]
this seems like not the actually intended behavior and i'd like to limit this to protection of the main branch.

Push access means i can't directly make a PR to start a discussion. I don't think that's the intention and what you want for now is more limited approval for actually merging things.

It isn't sustainable to say we won't allow discussion / include any updated protocols for months at a time - the alternative is that people will continue allocating codecs that make sense without contributing them or having the discussion here.

**Comment by rvagg** [comment]
@galargh could you help us with the settings to just protect merges into the main branch for now, then it's not so important who has branch access? I guess something similar to what's going on in go-car now might be appropriate and it looks like we have a bunch of settings to enable here to make that happen.

Alternatively, give me some pointers and I'll put up a PR.

**Comment by galargh** [comment]
I just enabled a more fine-grained configuration of branch protection rules in this repo.

Here's the current config for master protection of multicodecs: https://github.com/multiformats/github-mgmt/blob/c677062db48c79af7da8dea692958dbfd4587cee/github/multiformats.yml#L1055-L1070

Now you can modify them through a PR 🎉

**Comment by willscott** [comment]
Please also add back the `Contributors` group at least to be able to write to the repo

**Pull request merged**
